README.md

```diff
@@ -1,2 +1,45 @@
 # frequenz-resampling-rs
-Resampling library
+
+This project is the rust resampler for resampling a stream of samples to a given interval.
+
+## Usage
+
+To resample a vector of samples to a given interval, you can use the `Resampler` struct.
+The construction of a resampler expects an interval (`TimeDelta`) and a
+`ResamplingFunction`.
+Moreover, the `max_age` parameter can be used to control the maximum age of a sample.
+Setting it to 0 will remove all samples after resampling.
```

Comment:
Does this make sense @llucax?  We want `max_age >= 1` right?

Reply:
Yes, we have the functionality here that the interval that is multiplied with `max_age` adapts to the input interval, if the input interval is larger than the set `interval`.

```rust
self.input_interval = Some((s.timestamp() - input_start).max(self.interval));
```

Is that already what you have in mind, @llucax ?

There is also a test for this behavior: [a0262fea108af44a98cc49e5eb72641963dd816513f2d0a22eb738fcfed55ebeR201](https://github.com/frequenz-floss/frequenz-resampling-rs/pull/1/files#diff-a0262fea108af44a98cc49e5eb72641963dd816513f2d0a22eb738fcfed55ebeR201)

Reply:
I'm a bit confused. This seems to be talking about upsampling, right?

> Yes, we have the functionality here that the interval that is multiplied with max_age adapts to the input interval, if the input interval is larger than the set interval.
> 
> self.input_interval = Some((s.timestamp() - input_start).max(self.interval));
> 
> Is that already what you have in mind, @llucax ?

But this is about the zero `max_age`?

> There is also a test for this behavior: [a0262fea108af44a98cc49e5eb72641963dd816513f2d0a22eb738fcfed55ebeR201](https://github.com/frequenz-floss/frequenz-resampling-rs/pull/1/files#diff-a0262fea108af44a98cc49e5eb72641963dd816513f2d0a22eb738fcfed55ebeR201)

About the test, I'm not sure if I am misunderstanding the text above or the code, but that test seem to show that basically the resampling produces all samples with `None` as value when `max_age_in_intervals` is `0`. Is this correct? In the text above for me it sounds like the resampling is done with *something* (not sure what) and only then, after the resampling was done, the buffer is cleared out, which looks like the resampled samples shouldn't produce `None` in the test you linked.

The test looks OK, but I'm not sure if there is a point in allowing zero max age at all though, it ends up with `None` sample factory, right?